nautobot_firewall_models/models/core_models.py

```diff
@@ -656,3 +656,230 @@ def to_json(self):
+        row["original_destination_services"] = self.original_destination_services.all()
+        row["original_destination_service_groups"] = self.original_destination_service_groups.all()
+
+        row["translated_destination_address_groups"] = self.translated_destination_address_groups.all()
+        row["translated_destination_addresses"] = self.translated_destination_addresses.all()
+        row["translated_destination_services"] = self.translated_destination_services.all()
+        row["translated_destination_service_groups"] = self.translated_destination_service_groups.all()
+
+        row["mode"] = self.mode
+        row["log"] = self.log
+        row["status"] = self.status
+        row["request_id"] = self.request_id
+        return row
+
+    def to_json(self):
+        """Convenience method to convert to json."""
+        return model_to_json(self)
+
+    def __str__(self):
+        """Stringify instance."""
+        if self.request_id and self.name:
+            return f"{self.name} - {self.request_id}"
+        if self.name:
+            return self.name
+        return str(self.id)
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class NATPolicy(PrimaryModel):
+    """
+    The overarching model that is the full NAT policy with all underlying rules and child objects.
+
+    Each NATPolicy can be assigned to both devices and to dynamic groups which in turn can assign the policy to a related device.
+    """
+
+    description = models.CharField(
+        max_length=200,
+        blank=True,
+    )
+    name = models.CharField(max_length=100, unique=True)
+    nat_policy_rules = models.ManyToManyField(to=NATPolicyRule, through="NATPolicyRuleM2M", related_name="nat_policies")
+    assigned_devices = models.ManyToManyField(
+        to="dcim.Device", through="NATPolicyDeviceM2M", related_name="nat_policies"
+    )
+    assigned_dynamic_groups = models.ManyToManyField(
+        to="extras.DynamicGroup", through="NATPolicyDynamicGroupM2M", related_name="nat_policies"
+    )
+    status = StatusField(
+        on_delete=models.PROTECT,
+        related_name="%(app_label)s_%(class)s_related",  # e.g. dcim_device_related
+        default=get_default_status,
+    )
+    tenant = models.ForeignKey(
+        to="tenancy.Tenant",
+        on_delete=models.PROTECT,
+        related_name="nat_policies",
+        blank=True,
+        null=True,
+    )
+
+    class Meta:
+        """Meta class."""
+
+        ordering = ["name"]
+        verbose_name_plural = "NAT Policies"
+
+    def get_absolute_url(self):
+        """Return detail view URL."""
+        return reverse("plugins:nautobot_firewall_models:natpolicy", args=[self.pk])
+
+    def policy_details(self):
+        """Convenience method to convert to a Python list of dictionaries."""
+        data = []
```

Comment:
I believe you should be able to cast a queryset to a list.
```suggestion
        return list(self.nat_policy_rules.all())
```

Reply:
Improved this with a list comprehension